app/models/concerns/foreman_puppet/extensions/hostgroup.rb

```diff
@@ -35,6 +36,14 @@ def search_by_config_group(_key, operator, value)
           opts = "hostgroups.id IN(#{hostgroup_ids.join(',')})" if hostgroup_ids.present?
           { conditions: opts }
         end
+
+        def search_by_puppetclass(_key, operator, value)
+          conditions = sanitize_sql_for_conditions(["puppetclasses.name #{operator} ?", value_to_sql(operator, value)])
+          hostgroup_ids = ::Hostgroup.unscoped.with_taxonomy_scope.joins(puppet: :puppetclasses).where(conditions).map(&:subtree_ids)
+          conds = []
+          conds << "hostgroups.id IN (#{hostgroup_ids.join(',')})" if hostgroup_ids.present?
+          { conditions: conds.join(' OR ').presence || 'hostgroups.id < 0' }
```

Comment:
Nit: Empty line between `conds <<` and returning value

Reply:
Well, then let's do even more nitpicking and adjust also the formatting and naming scheme in lines 31ff?

Reply:
@stejskalleos If I look at this, I wonder why hostgroup search for `puppetclass` looks different from the one for `config_groups`.